execution-plane/arcon/Cargo.toml

```diff
@@ -25,34 +26,39 @@ arcon_error = { version = "0.1", path = "arcon_error" }
 arcon_extra = { version = "0.1", path = "arcon_extra" }
 kompact = { git = "https://github.com/kompics/kompact.git", branch = "master"}
 uuid = "0.8"
+static_assertions = "1.1.0"
+itertools = "0.8.2"
+smallbox = { version = "0.8", features = ["coerce"] }
+cfg-if = "0.1.10"
 fxhash = "0.2.1"
 
 # Serialisation
 abomonation = "0.7.3"
 abomonation_derive = "0.5.0"
 prost = { git = "https://github.com/Max-Meldrum/prost.git", branch = "derive_generics" }
 prost-types = "0.6"
+bytes = "0.5"
 
 # Optional
 rdkafka = { version = "0.23", optional = true }
 tokio = { version = "0.2", optional = true, features = ["full"] } # TODO: figure out which are truly needed
 tokio-util = { version = "0.2", optional = true, features = ["full"] }
-bytes = { version = "0.5", optional = true }
 futures = { version = "0.3", optional = true }
 serde_json = { version = "1.0.44", optional = true }
 serde = { version = "1.0.104", features = ["derive"], optional = true }
-rocksdb = { git = "https://github.com/tikv/rust-rocksdb.git", optional = true }
+bincode = { version = "1.2.1", optional = true }
+rocksdb = { git = "https://github.com/mrobakowski/rust-rocksdb.git", branch = "arcon", optional = true }
```

Comment:
I saw you mentioned that there are breaking changes in the PR on the rust-rocksdb repo. And the tests are not passing. Just concerned if this change will be integrated or not. I mean it is kinda the same with the prost branch. But the less custom branches the better :D 

If the changes are not merged, would it easy to adjust the current arcon code to work with the rust-rocksdb master?

Reply:
tikv version doesn't support checkpoints and here's my pull request for the one on crates.io https://github.com/rust-rocksdb/rust-rocksdb/pull/385. I _think_ it would be possible to make it work with the current version of rust-rocksdb, but it would have to use mutable global state and I don't really want to touch stuff like that.